### PR TITLE
Added simple fixed-value thresholding

### PR DESCRIPTION
**Pull request by Sythanis**
My first contribution to an open-source project. So I am new to a lot of stuff. I apologize if I made any mistakes. Tried to mimic the style as much as I can. Tested Threshold function on my project successfully. 

Also I hope I followed your contribution guidelines correctly. Got some nightly warnings but it should be fine.

If I can help in any other way please conntact me. 
Cheers,
Urban

### CODE REVIEW

File: src/core.rs [hunk 1]
```diff
@@ -427,3 +427,20 @@ pub enum NormType {
     /// Normalized using minmax distance
     MinMax = 32,
 }
+
+/// Threshold type. Please refer to [OpenCV's
+/// documentation] (https://docs.opencv.org/3.4.0/d7/d1b/group__imgproc__misc.html#gaa9e58d2860d4afa658ef70a9b1115576)
+#[repr(C)]
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Hash)]
+pub enum ThreshType {
```
Comment (nebgnahz) [inline src/core.rs]: Some suggestions:

1. s/ThreshType/ThresholdType
2. Use OpenCV trunk URL?
https://docs.opencv.org/trunk/d7/d1b/group__imgproc__misc.html#gaa9e58d2860d4afa658ef70a9b1115576
3. Each enum item can omit the prefix "THRESH". In this way, users would use `ThresholdType::Binary`, more clear than `ThreshType::ThreshBinary`.
4. Can you include all types? (and I think THRESH_BINARY is 0, not 1).

Comment (Sythanis) [reply src/core.rs]: Hey, thanks for the fast reply. Yea sorry, my bad regarding Threshhold types. Will change asap, aslo looking at the trunk URL.. should I add "Mask, Otsu and Triangle' flags to the enum as well. Cannot find what int's correspond to them. Probably just 5, 6, 7?

Comment (nebgnahz) [reply src/core.rs]: I think they are 7, 8, 16: https://github.com/opencv/opencv/blob/6e13ffe162b1e6ec79bcbea0dd609f33aef23051/modules/imgproc/include/opencv2/imgproc.hpp#L332

```
cv::ThresholdTypes { 
  cv::THRESH_BINARY = 0, 
  cv::THRESH_BINARY_INV = 1, 
  cv::THRESH_TRUNC = 2, 
  cv::THRESH_TOZERO = 3, 
  cv::THRESH_TOZERO_INV = 4, 
  cv::THRESH_MASK = 7, 
  cv::THRESH_OTSU = 8, 
  cv::THRESH_TRIANGLE = 16 
}
```

**Comment by nebgnahz** [comment]
Ping? I think this is almost ready to be merged (although my recent addition creates a merge conflict... but it is easily resolvable).

Are you still working on this?

**Comment by Sythanis** [comment]
Hey @nebgnahz Sorry compleatly forgot about this pull request. I will push tonight. Will have a look at the conflicts.
Will get back to you.

**Comment by nebgnahz** [comment]
Hi, @Sythanis Travis is failing because of format: https://travis-ci.org/nebgnahz/cv-rs/builds/369586993#L3621

Can you run `cargo fmt` and then push to this fork?

**Comment by Pzixel** [comment]
@Sythanis I suggest you to rebase on latest master - it may fix your issue

**Comment by Pzixel** [comment]
Because the PR author is missing for a quite long time I managed to fix all issues myself. Sorry for any inconvenience, if any.

**Comment by Pzixel** [comment]
There is already `cv_threshold` function which is implemented on mater branch:  https://nebgnahz.github.io/cv-rs/cv/mat/struct.Mat.html#method.threshold

**Pull request closed without merging**